### PR TITLE
General updates

### PR DESCRIPTION
**Pull request by pdmosses**
This PR aims to make the template easier to use for those new to Jekyll and the Just the Docs theme. The changes in this PR are merely my suggestions (based on my own lack of understanding of what terms like "deploy" really mean).

- Updates the home page at `index.md` to give an overview of *what* the template can do.
- Updates `README.md` to focus on *how* to do it ([preview](https://github.com/pdmosses/just-the-docs-template/blob/general-updates/README.md)).
- Minor polishing of `.gitignore`, `_config.yml`, and `Gemfile`.

When the template has been finalised, I'd like to add a reference to it on the theme home page.

### CODE REVIEW

**Comment by pdmosses** [comment]
@mattxwang I don't see how to request a review for this PR in the usual way…

**Review by mattxwang — APPROVED** [review]
No worries on the requesting review portion; this LGTM. Left one nit but we can address post-merge.

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    safe_yaml (1.0.5)
+    sassc (2.4.0)
+      ffi (~> 1.9)
+    terminal-table (3.0.2)
+      unicode-display_width (>= 1.1.1, < 3)
+    unicode-display_width (2.3.0)
+    webrick (1.7.0)
+
+PLATFORMS
+  x86_64-darwin-19
```
Comment (mattxwang) [inline Gemfile.lock:70]: 99% sure we may need to also eventually add other platforms, but we can do that separately!

**Comment by mattxwang** [comment]
> When the template has been finalised, I'd like to add a reference to it on the theme home page.

PR welcome for that as well!

File: Gemfile.lock (hunk 1 above)
Comment (mattxwang) [reply Gemfile.lock:70]: See: [88e466d](https://github.com/just-the-docs/just-the-docs-template/commit/88e466d5aa164ea0477c825ede4fb8f6ec658cb9)

Comment (pdmosses) [reply Gemfile.lock:70]: @mattxwang thanks for approving and merging this PR! I was expecting that you'd suggest some changes. Anyway, the template is something that we can easily update later, based on user feedback and experience.

When I next get a bit of spare time (hopefully this week) I'll try submitting a PR for the theme docs home page and README, referring to the template for details.

**Pull request merged**